src/mame/drivers/jpmsru.cpp

```diff
@@ -805,38 +1056,26 @@ void jpmsru_state::jpmsru_3k_busext(machine_config &config)
 void jpmsru_state::jpmsru_4k(machine_config &config)
 {
 	jpmsru_3k(config);
-}
 
-// Game configs
-void jpmsru_state::ewn(machine_config &config)
-{
-	jpmsru_3k_busext(config);
-	m_maincpu->set_addrmap(AS_IO, &jpmsru_state::outputs_ewn);
+	m_maincpu->set_addrmap(AS_PROGRAM, &jpmsru_state::jpmsru_4k_map);
 }
 
-void jpmsru_state::ewn2(machine_config &config)
-{
-	jpmsru_3k(config);
-	m_maincpu->set_addrmap(AS_IO, &jpmsru_state::outputs_ewn2);
-}
-
-void jpmsru_state::ndu(machine_config &config)
-{
-	jpmsru_3k_busext(config);
-	m_maincpu->set_addrmap(AS_IO, &jpmsru_state::outputs_ndu);
-}
-
-void jpmsru_state::dud(machine_config &config)
-{
-	jpmsru_3k(config);
-	m_maincpu->set_addrmap(AS_IO, &jpmsru_state::outputs_dud);
-}
-
-void jpmsru_state::lan(machine_config &config)
-{
-	jpmsru_3k(config);
-	m_maincpu->set_addrmap(AS_IO, &jpmsru_state::outputs_lan);
-}
+// Game configs
+#define SRU_MCONFIG(GAME, CONFIG)	void jpmsru_state::GAME(machine_config &config)\
+									{\
+										CONFIG(config);\
+										m_maincpu->set_addrmap(AS_IO, &jpmsru_state::outputs_##GAME);\
+									}
+
+SRU_MCONFIG( ewn, jpmsru_3k_busext )
+SRU_MCONFIG( ewn2, jpmsru_3k )
+SRU_MCONFIG( ndu, jpmsru_3k_busext )
+SRU_MCONFIG( dud, jpmsru_3k )
+SRU_MCONFIG( lan, jpmsru_3k )
+SRU_MCONFIG( ews, jpmsru_3k )
+SRU_MCONFIG( lt, jpmsru_4k )
+SRU_MCONFIG( sup2p, jpmsru_3k )
+SRU_MCONFIG( la, jpmsru_3k )
```

Comment:
This is less clear than just having the functions written out for minimal reduction in lines.  Also, `GAME` is a macro already, so using it as a parameter name may lead to confusion.

Reply:
Fixed.